src/annotator/anchoring/test/types-test.js

```diff
@@ -19,46 +21,335 @@ describe('types', function () {
+        const range = new RangeAnchor(container, new Range());
+        assert.deepEqual(range.toSelector({}), rangeSelectorResult());
+      });
+
+      it('returns a RangeSelector if options are missing', () => {
+        const range = new RangeAnchor(container, new Range());
+        assert.deepEqual(range.toSelector(), rangeSelectorResult());
+      });
+    });
+  });
+
+  describe('TextPositionAnchor', () => {
+    let fakePosFromRange;
+    let fakePosToRange;
+
+    beforeEach(() => {
+      fakePosFromRange = sinon.stub();
+      fakePosToRange = sinon.stub();
+      $imports.$mock({
+        'dom-anchor-text-position': {
+          fromRange: fakePosFromRange,
+          toRange: fakePosToRange,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+
+    function createTextPositionAnchor() {
+      return new TextPositionAnchor(container, 0, 0);
+    }
+
+    describe('#fromRange', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        fakePosFromRange.returns({
+          start: 0,
+          end: 1,
+        });
+        const anchor = TextPositionAnchor.fromRange(container, new Range());
+        assert.calledOnce(fakePosFromRange);
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#fromSelector', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        const anchor = TextPositionAnchor.fromSelector(container, {
+          start: 0,
+          end: 0,
+        });
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#toSelector', () => {
+      it('returns a TextPositionSelector', () => {
+        const anchor = createTextPositionAnchor();
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextPositionSelector',
+          start: 0,
+          end: 0,
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a range object', () => {
+        const anchor = createTextPositionAnchor();
+        fakePosToRange.returns('fake range');
+        assert.equal(anchor.toRange(), 'fake range');
+        assert.calledWith(fakePosToRange, container, { start: 0, end: 0 });
+      });
+    });
+
+    describe('integration tests', () => {
+      beforeEach(() => {
+        // restore dom-anchor-text-position to test third party lib integration
+        $imports.$restore();
+      });
+
+      it('can convert a Range to TextPositionSelector and back to a Range', () => {
+        const range = document.createRange();
+        range.setStart(container.firstChild, 0);
+        range.setEnd(container.firstChild, 4);
+        const anchor = TextPositionAnchor.fromRange(container, range);
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextPositionSelector',
+          start: 0,
+          end: 4,
+        });
+        const newRange = anchor.toRange();
+        assert.deepEqual(range, newRange);
```

Comment:
The arguments are the wrong way round here. The syntax is `assert.deepEqual(actual, expected)` and `range` is the _expected_ value while _newRange_ is the _actual_ value.